projects/shared/src/lib/paginator/paginator.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2021 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Subject } from 'rxjs';
+
+export class Paginator {
+
+  /** Event on more */
+  private _moreEvent = new Subject<number>();
+
+  private _id: string;
+
+  /** Records count */
+  private _recordsCount = 0;
+
+  /** Hidden info (after show more link) */
+  private _hiddenInfo: { singular: string, plurial: string } = undefined;
+
+  /** Hidden records count */
+  private _hiddenCount = 0;
+
+  /**
+   * Constructor
+   * @param _page - number (default: 1)
+   * @param _recordsPerPage - number (default: 10)
+   */
+  constructor(private _page: number = 1, private _recordsPerPage: number = 10) {}
+
+  /** More event */
+  moreEvent(): Subject<number> {
```

Comment:
change to a getter function or (better for me) change private to public

Reply:
I added a getter on event. This one returns an observable